integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/MessageContextFactory.java

```diff
@@ -0,0 +1,80 @@
+package no.difi.meldingsutveksling.noarkexchange;
+
+import no.difi.meldingsutveksling.MessageInformable;
+import no.difi.meldingsutveksling.ServiceIdentifier;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.domain.Avsender;
+import no.difi.meldingsutveksling.domain.Mottaker;
+import no.difi.meldingsutveksling.domain.Organisasjonsnummer;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookupException;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.ServiceRecord;
+import no.difi.meldingsutveksling.services.Adresseregister;
+import org.springframework.stereotype.Component;
+
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+
+@Component
+public class MessageContextFactory {
+
+    private IntegrasjonspunktProperties props;
+    private Adresseregister adresseregister;
+    private ServiceRegistryLookup serviceRegistryLookup;
+
+    public MessageContextFactory(IntegrasjonspunktProperties props,
+                                 Adresseregister adresseregister,
+                                 ServiceRegistryLookup srLookup) {
+        this.props = props;
+        this.adresseregister = adresseregister;
+        this.serviceRegistryLookup = srLookup;
+    }
+
+    public MessageContext from(MessageInformable message) throws MessageContextException {
+        MessageContext context = new MessageContext();
+        context.setAvsender(createAvsender(message.getSenderIdentifier()));
+        context.setMottaker(createMottaker(message.getReceiverIdentifier(), message.getServiceIdentifier()));
+        context.setJpId("");
+        context.setConversationId(message.getConversationId());
+        return context;
+    }
+
+    public MessageContext from(String senderOrgnr, String receiverOrgnr, String conversationId, Certificate certificate) throws MessageContextException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'no.difi.meldingsutveksling.noarkexchange.MessageContextException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)